github_api.rb

```diff
@@ -0,0 +1,41 @@
+require "rest-client"
+require "json"
+
+# GithubRepoApi Class
+# This class is here for communicating with the Github API to retrieve information
+# about a specific repo.
+#
+# Arguments:
+# username - The username to authenticate to github as (ex: user1234)
+# password - The password to quthenticate to github with (ex: password1234)
+# repo - The repository in github that you'd like to be getting info for (ex: Invoca/web)
+class GithubRepoApi
+  def initialize(username, password, repo)
+    @github_url = "https://#{username}:#{password}!@api.github.com/repos/#{repo}"
+  end
+
+  # Returns a hash containing the base information
+  # for the repository including but not limited to:
+  #
+  # fork - Boolean indicating if the repo was forked
+  def information!
+    JSON.parse(RestClient.get(@github_url))
+  end
+
+  # Returns a hash containing information about the current HEAD sha of the master branch
+  # including but not limited to:
+  #
+  # sha - The current sha at the head of master
+  def master_information!
+    JSON.parse(RestClient.get("#{@github_url}/git/refs/heads/master"))["object"]
+  end
```

Comment:
Same as above.
```suggestion
  def master_information
    JSON.parse(RestClient.get("#{@github_url}/git/refs/heads/master"))["object"]
  end
```
Or, make this method more specific:
```suggestion
  def master_sha
    JSON.parse(RestClient.get("#{@github_url}/git/refs/heads/master"))["object"]["sha"]
  end
```